File: 20.09.23/5_2.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main()
{
	int n;
	cin >> n;

	vector<vector<int>> dino;
	for (int i = 0; i < n; i++) {
		int x, y;
		cin >> x >> y;
		dino.push_back({ x, y, i+1 });
	}

	sort(dino.begin(), dino.end());

	for (int i = 0; i < n - 1; i++) {
		cout << dino[i][2] << ' ' << dino[i + 1][2] << '\n';
	}

	return 0;
}

File: 20.09.23/5_2.py
n = int(input())

dino = []
for i in range(n):
    x = int(input())
    y = int(input())
    dino.append([x, y, i+1])

dino.sort()

for i in range(n-1):
    print(dino[i][2], dino[i+1][2])
